ui/admin/app/routes/scopes/scope/credential-stores/credential-store/credential-libraries/new.js

```diff
@@ -52,16 +52,21 @@ export default class ScopesScopeCredentialStoresCredentialStoreCredentialLibrari
     const { id: credential_store_id } = this.modelFor(
       'scopes.scope.credential-stores.credential-store',
     );
+    let { name, description } = this.currentModel?.isNew
+      ? this.currentModel
+      : {};
 
     // Set the type to generic vault if feature flag isn't enabled in cases where
     // user sets the query parameter manually
-    if (!this.features.isEnabled('ssh-target')) {
-      type = TYPE_CREDENTIAL_LIBRARY_VAULT_GENERIC;
-    }
+    type = this.features.isEnabled('ssh-target')
+      ? type
+      : TYPE_CREDENTIAL_LIBRARY_VAULT_GENERIC;
```

Comment:
do we need this feature flag check at all? 

if the `type` is found in the model, set it else use the default type which is VAULT_GENERIC

Reply:
And we already set the default type on L51 😅  

this additional `type` reassignment seems redundant to me

Reply:
@DhariniJeeva agreed removing it

Reply:
hmm yeah, you are right. we would need it for that use case